### ISSUE TITLE
Control submenus messed up

### ISSUE THREAD

**Issue by Hexaae** [AFADoomer/Wolf3D-TC-3.0#11]
Control > Game Engine Options and Customize controls sub-menus are messed up...
![20221110_121605](https://user-images.githubusercontent.com/8591462/201077823-0da5ab87-4f57-474a-a839-f991162c1f7d.jpg)

I suppose many sub-menus could be simply hidden since many options are "hard-coded" for this total conversion and changing them can cause strange side-effects. It would probably wise to simplify menu subsystem for this Wolf3D TC...

**Comment by AFADoomer** [comment]
Ugh.  I hate the menu scaling code. What is your resolution and what are your various scaling options set to?

**Comment by Hexaae** [comment]
![20221110_123722](https://user-images.githubusercontent.com/8591462/201081184-b35868e0-123b-44dc-bead-8df013510f1e.jpg)
Everything else at default (never changed them)

**Comment by AFADoomer** [comment]
OK, that should be fixed as of 1e5f47f686c28ed07a81975264c80155c9d2f0b4.

There are still some other small issues with the menu code that I need to iron out...  And I have to remember to test in fullscreen mode.

**Comment by Hexaae** [comment]
Fixed, confirmed! 👍🏻

Would be possible to fix also 2 lines text overlapping the panel area? 😇
![Floor 1 - Wolfenstein 3D TC 11_11_2022 00_04_52](https://user-images.githubusercontent.com/8591462/201224836-0a169678-ed46-4c54-9324-2096140a8616.png)
![Floor 1 - Wolfenstein 3D TC 11_11_2022 00_04_06](https://user-images.githubusercontent.com/8591462/201224839-c26c6aea-50e1-4e3f-bea0-edff910472cd.png)

**Comment by AFADoomer** [comment]
Fixed the overdraw issue in 4147bc87e79535988a22a4abb8025fb1213fc5ca.

**Comment by Hexaae** [comment]
Uh, just noticed the "ENTER to change, BACKSPACE to clear" string in the customize controls is not centered.

**Comment by AFADoomer** [comment]
Fixed the text centering problem in 438063f5e264a8548da2083b7f8386c7ff726fc5.

I don't see any overlap in the load/save menu - either in fullscreen or windowed mode.  I assume you mean you see the frame/box contents overlapping with the black bar?

**Comment by Hexaae** [comment]
Yes, the Load/Save black bar in background gets overlapped by the standard listview frame of GZDoom...

**Comment by Hexaae** [comment]
Request: would be possible to use mouse to select options also in submenus instead of keyboad only?

**Comment by AFADoomer** [comment]
It works fine if you're not in fullscreen 😛 .

**Comment by AFADoomer** [comment]
Can you test with 0039c2837dab89d2fd6478d5cce12c726938673e?

**Comment by Hexaae** [comment]
Clicking now works but I'd expect two other things in submenus: upper right Back button, highlight items on mouse hover like main menu (with Pistol cursor following mouse hover).

**Comment by AFADoomer** [comment]
Can you set the 'disableautosave' cvar to 1 and try again?

**Comment by AFADoomer** [comment]
These should be fixed as of 8e70d4e4ddd71b2d69ca1a83292046bbb8e954bd.

**Comment by Hexaae** [comment]
![Wolfenstein 3D - Wolfenstein 3D TC 13_11_2022 01_47_52](https://user-images.githubusercontent.com/8591462/201501246-66841a31-bdad-481b-af38-b7bfa3a8ef63.png)
Strange char " '/ " at the end in last item.


![Wolfenstein 3D - Wolfenstein 3D TC 13_11_2022 01_59_06](https://user-images.githubusercontent.com/8591462/201501248-e12410fb-ae1d-497d-9f0f-4039ca141b6c.png)
When I use cursor keys Up/Down and reach the bottom of this menu, at some point it will disappear and you can hear just the select sound. Looks like some hidden items get selected but the listview won't scroll down as expected and stopped at Allow Suicide....

**Comment by AFADoomer** [comment]
> Strange char " '/ " at the end in last item.

This is the '%' sign from the line above

> When I use cursor keys Up/Down and reach the bottom of this menu, at some point it will disappear and you can hear just the select sound. Looks like some hidden items get selected but the listview won't scroll down as expected and stopped at Allow Suicide.... (1920x1080)

I'm aware of that one and able to reproduce, but no luck fixing yet.  It used to be far worse. 😄 

> Not yet fixed: on my 1920x1080 fullscreen I have to move mouse at screen bottom to see menu items highlighted on mouse hover... It's like the trigger is much above (Y) the actual pointer position.

Is this on all option menus, or just some?

**Comment by Hexaae** [comment]
> Is this on all option menus, or just some?

All sub-menus under Main Menu.

**Comment by AFADoomer** [comment]
Please test cursor hover/highlight handling in 6b792afadb681b22a18657948ab9f8ae5b99c47d.

**Comment by Hexaae** [comment]
Mouse hover is ok in sub-menus except for:
Player Setup
Gameplay Options
Compatibility Options

Still problems with cursor keys and some sub-menus like Gameplay Options:
![Wolfenstein 3D - Wolfenstein 3D TC 16_11_2022 02_23_25](https://user-images.githubusercontent.com/8591462/202060300-f763dd9c-baf8-441e-9ce8-ab109889fb34.png)

**Comment by Hexaae** [comment]
> > Strange char " '/ " at the end in last item.
> 
> This is the '%' sign from the line above

Can you make text strings larger to avoid that, moving selectors 50px+ on the right?

**Comment by AFADoomer** [comment]
Or you could play on a higher resolution or in windowed mode... 😛 

I'll adjust it slightly, but if I move it too much, I'll have to readjust other things, so...

**Comment by Hexaae** [comment]
Ok ;)

**Comment by AFADoomer** [comment]
OK, I think the scrolling issue with pressing up while at the top of the list is finally fixed in 4497f99e49a27c7fa2616736d4546b0dd5450b67.

**Comment by Hexaae** [comment]
Mmmh, not for:
- Player Setup
- Gameplay Options
- Compatibility Options
These sub-menus also suffer the mouse positioning problem (highlighting is much above the cursor position in 1920x1080 FS)

**Comment by Hexaae** [comment]
Another small bug: after you enter your name in the High Score the back button in the upper-left disappears.

**Comment by AFADoomer** [comment]
OK, as of abacd9ba7c0ad3da07d2311319b21862c3af7832 I think I've fixed the mouse positioning and back buttons now, and have added right click as a 'back' action to all menus.

The not scrolling after first loop is still broken.

**Comment by Hexaae** [comment]
> OK, as of [abacd9b](https://github.com/AFADoomer/Wolf3D-TC-3.0/commit/abacd9ba7c0ad3da07d2311319b21862c3af7832) I think I've fixed the mouse positioning and back buttons now, and have added right click as a 'back' action to all menus.

Confirmed working fine now.

**Comment by AFADoomer** [comment]
Menu scrolling issues should be resolved in 634de0c34b1b7dd4867f7b4dc5a744babd08dead.

**Comment by Hexaae** [comment]
Confirmed working fine. Thank you for your patience! 😊

**Comment by AFADoomer** [comment]
It's resolution dependent - I don't see the same issue as you do.  Annoying, but I don't know if I can reliably fix it easily.

**Comment by Hexaae** [comment]
Ok, just a very minor glitch detail...
